### ISSUE TITLE
[branch-2.8] Flaky-test: MultiTopicsReaderTest.testMultiTopic

### ISSUE THREAD

**Issue by codelipenghui** [apache/pulsar#13698]
https://github.com/apache/pulsar/runs/4763566154?check_suite_focus=true

```
  Error:  Tests run: 33, Failures: 1, Errors: 0, Skipped: 27, Time elapsed: 44.085 s <<< FAILURE! - in org.apache.pulsar.client.impl.MultiTopicsReaderTest
  Error:  testMultiTopic(org.apache.pulsar.client.impl.MultiTopicsReaderTest)  Time elapsed: 0.429 s  <<< FAILURE!
  java.lang.AssertionError: expected [0] but found [5]
  	at org.testng.Assert.fail(Assert.java:99)
  	at org.testng.Assert.failNotEquals(Assert.java:1037)
  	at org.testng.Assert.assertEqualsImpl(Assert.java:140)
  	at org.testng.Assert.assertEquals(Assert.java:122)
  	at org.testng.Assert.assertEquals(Assert.java:907)
  	at org.testng.Assert.assertEquals(Assert.java:917)
  	at org.apache.pulsar.client.impl.MultiTopicsReaderTest.testMultiTopic(MultiTopicsReaderTest.java:437)
  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
  	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:132)
  	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:45)
  	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:73)
  	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
  	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
  	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
  	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
  	at java.base/java.lang.Thread.run(Thread.java:829)
```

**Comment by mattisonchao** [comment]
@codelipenghui  i can work on it.

**Comment by Demogorgon314** [comment]
@codelipenghui May be fixed by #13634
